test/dependency-manager/dependencies.test.ts

```diff
@@ -0,0 +1,504 @@
+        status: 200,
+        response: { tag: 'v1', config: { name: 'architect/payments-service', parameters: { WORKED: 0 } }, service: { url: 'architect/payments-service:v1' } }
+      });
+
+      const env_config = {
+        services: {
+          'architect/checkout-service:latest': {}
+        }
+      };
+
+      mock_fs({
+        '/stack/arc.env.json': JSON.stringify(env_config),
+      });
+
+      const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+      const graph = manager.graph;
+      expect(graph.nodes).length(2);
+      expect(graph.edges).length(1);
+
+      expect((graph.nodes[0] as ServiceNode).node_config.getParameters()).keys(['WORKED']);
+      expect((graph.nodes[0] as ServiceNode).node_config.getParameters().WORKED.default).eq(1);
+      expect((graph.nodes[1] as ServiceNode).node_config.getParameters()).keys(['WORKED']);
+      expect((graph.nodes[1] as ServiceNode).node_config.getParameters().WORKED.default).eq(1);
+    });
+
+    it('load service and then load dependencies with override in env config', async () => {
+      const service_config = {
+        parameters: { WORKED: 1 },
+        dependencies: {
+          'architect/payments-service': {
+            ref: 'v1',
+            parameters: { WORKED: 1 }
+          }
+        }
+      }
+      moxios.stubRequest(`/accounts/architect/services/checkout-service/versions/latest`, {
+        status: 200,
+        response: { tag: 'latest', config: service_config, service: { url: 'architect/checkout-service:latest' } }
+      });
+
+      moxios.stubRequest(`/accounts/architect/services/payments-service/versions/v1`, {
+        status: 200,
+        response: { tag: 'v1', config: { name: 'architect/payments-service', parameters: { WORKED: 0 } }, service: { url: 'architect/payments-service:v1' } }
+      });
+
+      const env_config = {
+        services: {
+          'architect/checkout-service:latest': {
+            dependencies: {
+              'architect/payments-service': {
+                parameters: { WORKED: 2 }
+              }
+            }
+          }
+        }
+      };
+
+      mock_fs({
+        '/stack/arc.env.json': JSON.stringify(env_config),
+      });
+
+      const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+      const graph = manager.graph;
+      expect(graph.nodes).length(2);
+      expect(graph.edges).length(1);
+
+      expect((graph.nodes[0] as ServiceNode).node_config.getParameters()).keys(['WORKED']);
+      expect((graph.nodes[0] as ServiceNode).node_config.getParameters().WORKED.default).eq(1);
+      expect((graph.nodes[1] as ServiceNode).node_config.getParameters()).keys(['WORKED']);
+      expect((graph.nodes[1] as ServiceNode).node_config.getParameters().WORKED.default).eq(2);
+    });
+
+    /*
+    it('chained refs', async () => {
```

Comment:
@davidthor is thing something that you want to support?

Reply:
I would expect that to work, yes.

Reply:
jerk

Reply:
haha, aside from it being more work for you, is there a reason not to? Seems like something I would expect to work if I were a user.

Reply:
yeah I'd expect it to work - just requires a while loop of api calls until we stop getting refs